client/src/main/java/io/pravega/client/stream/notifications/notifier/ScaleEventNotifier.java

```diff
@@ -81,10 +83,15 @@ public String getType() {
     private void checkAndTriggerScaleNotification() {
         this.synchronizer.fetchUpdates();
         ReaderGroupState state = this.synchronizer.getState();
-        ScaleEvent event = ScaleEvent.builder().numOfSegments(state.getNumberOfSegments())
-                         .numOfReaders(state.getOnlineReaders().size())
-                         .build();
-        if (event.getNumOfReaders() > 0) { //Notify only if there are online readers in the reader group.
+        int newNumberOfSegments = state.getNumberOfSegments();
+
+        if (this.numberOfSegments == 0) {
```

Comment:
Event though it may sound obvious, it is always good practice to write down a comment and say that we are doing this for initialization purposes. I suppose that one invariant here is that this happens only upon the first call and the value of `this.numberOfSegments` can never be zero again. It would be good if we could assert that this invariant actually holds.

Reply:
Done. Added the invariant check and comment.